### PR TITLE
Fix operation stats for map and replicatedMap

### PR DESCRIPTION
**Pull request by sancar**
Map.clear and Map.containsKey, ReplicatedMap.get were not updating
the stats.
ReplicatedMap.clear were updating the stats twice.

Fixed the issues above and added tests for member and client.

fixes https://github.com/hazelcast/hazelcast/issues/14143

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/map/MapGetAllMessageTask.java [hunk 1]
```diff
@@ -53,6 +55,22 @@ public String getServiceName() {
...
+    @Override
+    protected void beforeProcess() {
+        startTimeNanos = System.nanoTime();
+    }
+
+    @Override
+    protected void beforeResponse() {
+        final long latencyNanos = System.nanoTime() - startTimeNanos;
+        final MapService mapService = getService(MapService.SERVICE_NAME);
+        MapContainer mapContainer = mapService.getMapServiceContext().getMapContainer(parameters.name);
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/map/MapGetAllMessageTask.java]: same thread visibility issue may occur here with `parameters.name` and `parameters.keys.size()` below?

Comment (sancar) [reply hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/map/MapGetAllMessageTask.java]: These values are created before the thread is started and does not change after that. They are immutable.
So they should be ok.

File: hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/map/MapGetAllMessageTask.java [hunk 2]
```diff
@@ -33,6 +34,7 @@
 public class MapGetAllMessageTask
         extends AbstractPartitionMessageTask<MapGetAllCodec.RequestParameters> {
 
+    private long startTimeNanos;
```
[the remark this answers is not in the document]
Comment (sancar) [reply hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/map/MapGetAllMessageTask.java]: Added volatile to 3 places in total together with this.

**Comment by vbekiaris** [comment]
run-lab-run

**Comment by sancar** [comment]
run-lab-run

**Pull request merged**